tests/scripts/all.sh

```diff
@@ -2878,20 +2879,22 @@ component_build_armcc () {
 
     make clean
 
+    # Compile with -O1 since some Arm inline assembly is disabled for -O0.
+
     # ARM Compiler 6 - Target ARMv7-A
-    armc6_build_test "--target=arm-arm-none-eabi -march=armv7-a"
+    armc6_build_test "--target=arm-arm-none-eabi -O1 -march=armv7-a"
```

Comment:
Minor: I'd prefer to have `--target` and `-march` next to each other since they're strongly related.